cluster-autoscaler/cloudprovider/azure/azure_agent_pool.go

```diff
@@ -473,7 +473,30 @@ func (as *AgentPool) DeleteNodes(nodes []*apiv1.Node) error {
 
 // ForceDeleteNodes deletes nodes from the group regardless of constraints.
 func (as *AgentPool) ForceDeleteNodes(nodes []*apiv1.Node) error {
```

Comment:
To avoid code duplication, DeleteNodes should delegate to ForceDeleteNodes after checking constraints